### ISSUE TITLE
Hidden power pins in 74xGxx library (KLC 4.7.ii)

### ISSUE THREAD

**Issue by RileyStarlight** [KiCad/kicad-symbols#756]
I have recently updated the schematics libraries and found that the new symbols for the 74xGxx family have hidden power pins. This makes it absolutely incompatible with mixed voltage environments, and from what I've been seeing, it also goes against KLC.

I'm using the latest libraries for kicad 4 repository, but the ones in this repo are identical.

**Comment by thomasonw** [comment]
+1 on this issue.  Embed and per-defined and hidden forcing of power and ground to Vcc and GND is IMHO very poor form.  Precludes distributed power / ground designed and I would offer this really needs to be addressed before doing Ver 5 release.

**Comment by jcyrax** [comment]
I've started work on this issue (PR:https://github.com/KiCad/kicad-symbols/pull/371), but due to time constraint its unfinished. Symbol part is basically done, but my PR needs review and maybe some further changes based on that. I have missing footprints created, but need to make PR for them.

**Comment by evanshultz** [comment]
@jcyrax 
Have you pushed the symbols? And did you script them (symgen or other)?

If the footprints are already broken now, adding power pins would still be an upgrade with no drawback and potentially able to merge right away IMO.

**Comment by jcyrax** [comment]
Yes, the latest changes are in my PR (https://github.com/KiCad/kicad-symbols/pull/371). They are scripted with symgen.

**Comment by henryhallam** [comment]
Any movement on this? Personally I don't care whether the power pins are a separate unit symbol in a box, separate unit symbol as "floating pins", or on the first unit symbol - but the present situation with them being hidden pins is highly unsatisfactory.

**Comment by kwendenarmo** [comment]
I think that each unit has its own power pins is most appropriate. And so it is in for example op-amps, both multipart and single.

**Comment by poeschlr** [comment]
My vote is on having power pins as a separate unit as it will offer the flexibility to support all usecases.
I am not really sure if we can include such a change in 5.1 or if we need to wait till v6. We might want to ask the devs about that.

Regarding the pin count filter: I am not so sure that the filter is broken by having pins on all units. (They should not be counted multiple times as long as the pins are set as common to all units.)

There even is a ERC check now that reduces the chance of connecting multple nets to different duplicates of the same pin. (However there are also some bug reports indicating that it does not quite work right now.)

**Comment by poeschlr** [comment]
I have a solution that would even work if the devs do not want us to update the current symbols. We could make a new library that holds the fixed footprints. We would then delete the old one from the library table but keep the old file.
Right now i have no idea what that lib could be called but that is something we can decide later on.

**Comment by antoniovazquezblanco** [comment]
@poeschlr can we get a decision on what option should be taken to fix the issue and what release version will be targeted?

Thanks!

**Comment by aewallin** [comment]
74132 part listed here:
https://forum.kicad.info/t/hidden-pins-upper-and-lower-case-problem/14199

maybe it's worth splitting this issue into smaller pieces of work?

**Comment by poeschlr** [comment]
These should be scripted anyways. Take a look at https://github.com/bobc/kicad-symgen/tree/master/symgen

**Comment by calebreister** [comment]
When I was working on the [HSMC_Card symbols](https://github.com/KiCad/kicad-symbols/pull/1428), I used Awk to perform quick bulk edits to pins. Perhaps a sufficiently complex Awk script could also be used solve this problem (and I don't mean a one-liner).

**Comment by calebreister** [comment]
This issue also holds true for the 74xx library.

**Comment by evanshultz** [comment]
There are a number of issues and a PR about the logic libs, but this one has the most recent comments so I'll tag on here.

Because of the symbol format change in v6, I propose to put this on hold until that is implemented. Then revamp the logic libraries for v6. Agree? Disagree? Other ideas?

**Comment by calebreister** [comment]
Sounds like a good idea to me. Is there any documentation available for the new library format?

**Comment by evanshultz** [comment]
https://lists.launchpad.net/kicad-developers/msg38760.html

Warning, it goes off the rails quickly...
